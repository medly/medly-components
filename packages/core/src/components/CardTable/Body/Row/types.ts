import { WithThemeProp } from '@medly-components/utils';

export interface Props extends WithThemeProp {
    onClick?: () => void;
    isRowClickDisabled?: boolean;
    withWhiteBackground?: boolean;
    gridTemplateColumns: string;
}
