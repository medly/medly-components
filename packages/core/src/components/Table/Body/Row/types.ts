import { HTMLProps, Omit, WithThemeProp } from '@medly-components/utils';
import { GridTemplateProps } from '../../types';

export type StyledProps = GridTemplateProps &
    HTMLProps<HTMLTableRowElement> &
    WithThemeProp & {
        isSelected?: boolean;
        isExpanded?: boolean;
        isExpandable?: boolean;
        showRowWithCardStyle?: boolean;
        withRowSeparators?: boolean;
    };
export type RowProps = Omit<HTMLProps<HTMLTableRowElement>, 'style' | 'data'> & {
    data: { [key: string]: any };
    selectedRowIds: Array<any>;
    onRowSelection: (id: any) => void;
    showShadowAfterFrozenElement?: boolean;
};
