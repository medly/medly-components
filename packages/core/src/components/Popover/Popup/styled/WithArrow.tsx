import { css } from 'styled-components';
import { PopoverPopupProps } from '../types';

const getPositionWithArrow = ({ placement }: PopoverPopupProps) => {
    switch (placement) {
        case 'top':
            return css`
                box-shadow: 0 0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: translateX(-50%);
                    top: 100%;
                    left: 50%;
                }
            `;
        case 'top-start':
            return css`
                box-shadow: 0 0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    top: 100%;
                    left: 2rem;
                }
            `;
        case 'top-end':
            return css`
                box-shadow: 0 0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    top: 100%;
                    right: 2rem;
                }
            `;
        case 'right':
            return css`
                box-shadow: -0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: translateY(-50%) rotate(90deg);
                    top: 50%;
                    right: calc(100% - 0.3rem);
                }
            `;
        case 'right-start':
            return css`
                box-shadow: -0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(90deg);
                    top: 2rem;
                    right: calc(100% - 0.3rem);
                }
            `;
        case 'right-end':
            return css`
                box-shadow: -0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(90deg);
                    bottom: 2rem;
                    right: calc(100% - 0.3rem);
                }
            `;
        case 'bottom':
            return css`
                box-shadow: 0 -0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: translateX(-50%) rotate(180deg);
                    left: 50%;
                    bottom: 100%;
                }
            `;
        case 'bottom-start':
            return css`
                box-shadow: 0 -0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(180deg);
                    left: 2rem;
                    bottom: 100%;
                }
            `;
        case 'bottom-end':
            return css`
                box-shadow: 0 -0.4rem 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(180deg);
                    right: 2rem;
                    bottom: 100%;
                }
            `;
        case 'left':
            return css`
                box-shadow: 0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: translateY(-50%) rotate(270deg);
                    top: 50%;
                    left: calc(100% - 0.3rem);
                }
            `;
        case 'left-start':
            return css`
                box-shadow: 0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(270deg);
                    top: 2rem;
                    left: calc(100% - 0.3rem);
                }
            `;
        case 'left-end':
            return css`
                box-shadow: 0.4rem 0 1.6rem rgba(176, 188, 200, 0.5);
                &::after {
                    transform: rotate(270deg);
                    bottom: 2rem;
                    left: calc(100% - 0.3rem);
                }
            `;
    }
};

export const getStyleWithArrow = css<PopoverPopupProps>`
    border-radius: 0.8rem;
    padding: 1.5rem 2rem;

    &::after {
        content: '';
        width: 0;
        height: 0;
        z-index: 4;
        position: absolute;
        border-left: 0.8rem solid transparent;
        border-right: 0.8rem solid transparent;
        border-top: 1rem solid ${({ bgColor }) => bgColor};
    }

    ${getPositionWithArrow}
`;
