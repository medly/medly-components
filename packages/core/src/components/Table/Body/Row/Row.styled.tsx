import { css, styled } from '@medly-components/utils';
import { rgba } from 'polished';
import { GridTemplateProps } from '../../types';
import { StyledProps } from './types';

const cardStyle = css<StyledProps>`
    border: none;
    border-radius: 0.8rem;
    margin-left: 0.8rem;
    margin-right: 0.8rem;
    box-shadow: ${({ theme }) => `0 0.2rem 0.8rem ${rgba(theme.table.shadowColor, 0.2)} `};
    &,
    & > * {
        background-color: ${({ theme, isSelected, disabled }) =>
            theme.table.row.bgColor[disabled ? 'disabled' : isSelected ? 'selected' : 'even']};
    }

    &&& > * {
        &:first-child {
            &,
            &::before {
                transition: ${props => !props.isExpanded && 'border-radius 200ms ease-in'};
                border-radius: ${props => (props.isExpanded ? '0.8rem 0 0 0' : '0.8rem 0 0 0.8rem')};
            }

            &::before {
                width: 0.8rem;
            }
        }
        &:last-child {
            border-radius: ${props => (props.isExpandable ? '0 0 0.8rem 0' : '0 0.8rem 0.8rem 0')};
        }
        &:nth-last-child(3) {
            border-radius: ${props => props.isExpandable && (props.isExpanded ? '0 0.8rem 0 0' : '0 0.8rem 0.8rem 0')};
        }
        &:nth-last-child(2) {
            border-radius: ${props => props.isExpandable && '0 0 0 0.8rem'};
            &::before {
                width: 0.8rem;
                border-radius: 0 0 0 0.8rem;
            }
        }
    }

    &:hover {
        box-shadow: ${({ theme }) => `0 0.2rem 0.8rem ${rgba(theme.table.shadowColor, 0.35)} `};
    }

    &:not(:last-child) {
        margin-bottom: 0.8rem;
    }
`;

const normalStyle = css<StyledProps>`
    &:hover {
        z-index: 2;
        box-shadow: ${({ disabled, onClick, theme }) => !disabled && onClick && `0 0.2rem 0.4rem ${rgba(theme.table.shadowColor, 0.2)} `};
    }

    &:nth-child(odd) {
        &,
        & > * {
            background-color: ${({ theme, isSelected, disabled }) =>
                theme.table.row.bgColor[disabled ? 'disabled' : isSelected ? 'selected' : 'odd']};
        }
    }

    &:nth-child(even) {
        &,
        & > * {
            background-color: ${({ theme, isSelected, disabled }) =>
                theme.table.row.bgColor[disabled ? 'disabled' : isSelected ? 'selected' : 'even']};
        }
    }

    &:not(:last-child) {
        border-bottom: 1px solid ${({ theme }) => theme.table.borderColor};
    }
`;

export const Row = styled('tr').attrs(({ gridTemplateColumns }: GridTemplateProps) => ({
    style: {
        gridTemplateColumns
    }
}))<StyledProps>`
    display: grid;
    position: relative;
    align-items: center;
    min-width: fit-content;
    cursor: ${({ disabled, onClick }) => (disabled ? 'not-allowed' : onClick ? 'pointer' : 'inherit')};
    ${({ showRowWithCardStyle }) => (showRowWithCardStyle ? cardStyle : normalStyle)}
`;

const noResultCardStyle = () => {
    return css`
        padding: 1.5rem 0;
    `;
};

export const NoResultRow = styled(Row)`
    justify-content: center;
    align-items: center;
    ${({ showRowWithCardStyle }) => showRowWithCardStyle && noResultCardStyle}
`;

export const NoResultCell = styled('td')`
    grid-column: 1/-1;
`;
