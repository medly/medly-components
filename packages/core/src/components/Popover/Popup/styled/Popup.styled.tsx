import { fullHeight, fullWidth } from '@medly-components/utils';
import styled, { css } from 'styled-components';
import { Placement } from '../../types';
import { StyledPopoverPopupProps } from '../types';
import { getStyleWithArrow } from './WithArrow';

export const getPosition = ({ placement, distance = '0px' }: { placement: Placement; distance?: string }) => {
    switch (placement) {
        case 'top-start':
            return css`
                bottom: calc(100% + ${distance});
                left: 0%;
            `;
        case 'top':
            return css`
                bottom: calc(100% + ${distance});
                left: 50%;
                transform: translateX(-50%);
            `;
        case 'top-end':
            return css`
                bottom: calc(100% + ${distance});
                right: 0%;
            `;
        case 'right-start':
            return css`
                top: 0%;
                left: calc(100% + ${distance});
            `;
        case 'right':
            return css`
                left: calc(100% + ${distance});
                top: 50%;
                transform: translateY(-50%);
            `;
        case 'right-end':
            return css`
                bottom: 0;
                left: calc(100% + ${distance});
            `;
        case 'bottom-start':
            return css`
                top: calc(100% + ${distance});
                left: 0%;
            `;
        case 'bottom':
            return css`
                top: calc(100% + ${distance});
                left: 50%;
                transform: translateX(-50%);
            `;
        case 'bottom-end':
            return css`
                top: calc(100% + ${distance});
                right: 0%;
            `;
        case 'left-start':
            return css`
                right: calc(100% + ${distance});
                top: 0%;
            `;
        case 'left':
            return css`
                right: calc(100% + ${distance});
                top: 50%;
                transform: translateY(-50%);
            `;
        case 'left-end':
            return css`
                right: calc(100% + ${distance});
                bottom: 0%;
            `;
    }
};

export const PopupStyled = styled('div')<StyledPopoverPopupProps>`
    z-index: 4;
    position: absolute;
    background-color: ${({ bgColor }) => bgColor};

    ${props => props.fullWidth && fullWidth()};
    ${props => props.fullHeight && fullHeight()};

    ${props => getPosition({ placement: props.placement, distance: props.withArrow ? '1.5rem' : props.distance })}
    ${({ withArrow }) => withArrow && getStyleWithArrow}
`;
