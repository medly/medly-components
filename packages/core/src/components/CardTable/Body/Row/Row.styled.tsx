import styled from 'styled-components';
import { Props } from './types';

export const Row = styled('tr').attrs(({ withWhiteBackground, theme: { cardTable } }: Props) => ({
    bgTheme: withWhiteBackground ? 'lightBackground' : 'darkBackground',
    cardTable
}))<Props & { bgTheme: 'lightBackground' | 'darkBackground' }>`
    display: grid;
    grid-template-columns: ${({ gridTemplateColumns }) => gridTemplateColumns};
    cursor: ${({ onClick, isRowClickDisabled }) => (onClick && !isRowClickDisabled ? 'pointer' : 'inherit')};
    box-sizing: border-box;
    border-radius: 0.8rem;
    margin-bottom: 0.8rem;
    background-color: white;
    transition: box-shadow 100ms ease-out;
    border: 0.1rem solid ${({ bgTheme, cardTable }) => cardTable[bgTheme].borderColor};
    box-shadow: ${({ bgTheme, cardTable }) => cardTable[bgTheme].boxShadow};

    &:hover {
        box-shadow: ${({ bgTheme, cardTable }) => cardTable[bgTheme].hoverBoxShadow};
    }

    :nth-child(odd) {
        background: ${({ bgTheme, cardTable, isRowClickDisabled }) =>
            !isRowClickDisabled ? cardTable[bgTheme].oddRowBgColor : cardTable.disabledRowBgColor};
    }
    :nth-child(even) {
        background: ${({ bgTheme, cardTable, isRowClickDisabled }) =>
            !isRowClickDisabled ? cardTable[bgTheme].evenRowBgColor : cardTable.disabledRowBgColor};
    }
`;

export const NoResult = styled(Row)`
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 1.5rem 0;
`;
