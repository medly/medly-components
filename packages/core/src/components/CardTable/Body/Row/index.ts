import { Row } from './Row.styled';

export default Row;
