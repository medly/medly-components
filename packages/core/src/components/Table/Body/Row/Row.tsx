import type { FC, KeyboardEvent } from 'react';
import { memo, useCallback, useContext, useEffect, useMemo, useRef, useState } from 'react';
import { TableComponentsCommonPropsContext } from '../../context';
import { GroupCell } from '../../GroupCell';
import { getGridTemplateColumns, getNestedValue } from '../../helpers';
import { TableColumnConfig } from '../../types';
import Cell from '../Cell';
import ExtendedRowCell from '../Cell/ExtendedRowCell';
import RowActionsCell from '../Cell/RowActionsCell';
import RowHoverActionsCell from '../Cell/RowHoverActionsCell';
import * as Styled from './Row.styled';
import { RowProps } from './types';

export const Row: FC<RowProps> = memo(props => {
    const [isRowHovered, setIsRowHovered] = useState(false),
        {
            id,
            data,
            isNavigated = false,
            isRowExpandedFromKeyboard,
            isRowCollapsedFromKeyboard,
            showShadowAfterFrozenElement,
            selectedRowIds,
            onRowSelection,
            ...restProps
        } = props,
        {
            keyBindings,
            columns,
            isLoading,
            onRowClick,
            hiddenDivRef,
            addColumnMaxSize,
            rowClickDisableKey,
            rowSelectionDisableKey,
            defaultRowExpandKey,
            isRowSelectable,
            isRowExpandable,
            isGroupedTable,
            showRowWithCardStyle,
            size: tableSize,
            expandedRowComponent,
            withRowSeparators,
            rowHoverActions: RowHoverActions
        } = useContext(TableComponentsCommonPropsContext);

    const [isExpanded, setExpansionState] = useState<boolean>(defaultRowExpandKey && data[defaultRowExpandKey]),
        isRowSelected = useMemo(() => !isLoading && selectedRowIds.includes(id), [id, isLoading, selectedRowIds]),
        isRowClickDisabled = useMemo(() => rowClickDisableKey && data[rowClickDisableKey], [data, rowClickDisableKey]),
        isRowSelectionDisabled = useMemo(() => rowSelectionDisableKey && data[rowSelectionDisableKey], [data, rowSelectionDisableKey]),
        handleRowSelection = useCallback(() => onRowSelection(id), [id, onRowSelection]),
        handleExpansionIconClick = useCallback(() => setExpansionState(val => !val), []),
        handleRowClick = useMemo(
            () => (!isLoading && onRowClick && !isRowClickDisabled ? () => onRowClick(data) : undefined),
            [isLoading, data, onRowClick, isRowClickDisabled, isRowExpandable, handleExpansionIconClick]
        ),
        handleMouseEnter = useCallback(() => setIsRowHovered(true), []),
        handleMouseLeave = useCallback(() => setIsRowHovered(false), []),
        ref = useRef<HTMLTableRowElement>(null);

    const getCells = useCallback(
        (rowData: any = {}, configs: TableColumnConfig[] = columns, field = '') =>
            configs.reduce((cells, config) => {
                const fieldName = `${field && `${field}.`}${config.field}`;

                return config.field === 'row-actions'
                    ? cells
                    : [
                          ...cells,
                          config.children ? (
                              <GroupCell
                                  as={field ? 'div' : 'td'}
                                  key={config.field}
                                  hidden={config.hidden}
                                  gridTemplateColumns={getGridTemplateColumns(config.children)}
                              >
                                  {getCells(rowData[config.field], config.children, config.field)}
                              </GroupCell>
                          ) : (
                              <Cell
                                  rowId={id}
                                  key={fieldName}
                                  config={config}
                                  as={field ? 'div' : 'td'}
                                  isLoading={isLoading}
                                  rowData={rowData}
                                  data={getNestedValue(rowData, config.field)}
                                  addColumnMaxSize={addColumnMaxSize}
                                  isRowClickDisabled={isRowClickDisabled}
                                  dottedFieldName={fieldName}
                                  tableSize={tableSize}
                                  hiddenDivRef={hiddenDivRef}
                              />
                          )
                      ];
            }, [] as React.ReactElement[]),
        [id, data, isLoading, columns, addColumnMaxSize, isRowClickDisabled, hiddenDivRef, tableSize]
    );

    useEffect(() => {
        if (isNavigated) {
            if (isRowSelectable && !isRowSelectionDisabled) ref.current?.querySelector('input')?.focus();
            else ref.current?.focus();
        }
    }, [isNavigated]);

    useEffect(() => {
        isRowExpandedFromKeyboard && setExpansionState(true);
    }, [isRowExpandedFromKeyboard]);

    useEffect(() => {
        isRowCollapsedFromKeyboard && setExpansionState(false);
    }, [isRowCollapsedFromKeyboard]);

    // TODO: Check why useKeypress is not working in this case
    const handleRowClickFromKeyboard = useCallback(
        (e: KeyboardEvent<HTMLTableRowElement>) => {
            e.key === keyBindings.rowClick! && isNavigated && handleRowClick && handleRowClick();
        },
        [isNavigated, handleRowClick]
    );

    return (
        <>
            <Styled.Row
                {...restProps}
                ref={ref}
                tabIndex={Number(id)}
                disabled={isRowClickDisabled}
                onClick={handleRowClick}
                isSelected={isRowSelected}
                isExpanded={isExpanded}
                isExpandable={isRowExpandable}
                showRowWithCardStyle={showRowWithCardStyle}
                gridTemplateColumns={getGridTemplateColumns(columns)}
                onMouseEnter={handleMouseEnter}
                onMouseLeave={handleMouseLeave}
                withRowSeparators={withRowSeparators}
                isNavigated={isNavigated}
                onKeyDown={handleRowClickFromKeyboard}
            >
                {(isRowSelectable || isRowExpandable) && (
                    <RowActionsCell
                        tableSize={tableSize}
                        isLoading={isLoading}
                        isRowExpanded={isExpanded}
                        isGroupedTable={isGroupedTable}
                        isRowExpandable={isRowExpandable}
                        isRowSelected={isRowSelected}
                        isRowSelectable={isRowSelectable}
                        isRowSelectionDisabled={isRowSelectionDisabled}
                        onRowSelection={handleRowSelection}
                        onRowExpansionIconClick={handleExpansionIconClick}
                        showShadowAtRight={showShadowAfterFrozenElement}
                        isNavigated={isNavigated}
                    />
                )}
                {getCells(data)}
                {isRowExpandable && expandedRowComponent && (
                    <ExtendedRowCell
                        rowId={id!}
                        rowData={data}
                        tableSize={tableSize}
                        isRowExpanded={isExpanded}
                        isRowSelected={isRowSelected}
                        isGroupedTable={isGroupedTable}
                        expandedRowComponent={expandedRowComponent}
                        isRowClickDisabled={isRowClickDisabled}
                        showShadowAtRight={showShadowAfterFrozenElement}
                    />
                )}
                {RowHoverActions !== undefined && !isRowClickDisabled && !isLoading && (
                    <RowHoverActionsCell show={isRowHovered}>
                        <RowHoverActions rowData={data} rowId={id} />
                    </RowHoverActionsCell>
                )}
            </Styled.Row>
        </>
    );
});
Row.displayName = 'Row';
