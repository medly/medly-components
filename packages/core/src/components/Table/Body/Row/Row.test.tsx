import { render } from '@test-utils';
import React from 'react';
import { NoResultRow, NoResultCell, Row } from './Row.styled';

describe('Row', () => {
    it('should render row properly', () => {
        const { container } = render(<Row>Dummy</Row>);
        expect(container).toMatchSnapshot();
    });

    it('should render NoResult properly', () => {
        const { container } = render(
            <NoResultRow>
                <NoResultCell>Dummy</NoResultCell>
            </NoResultRow>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render NoResult with Card design also', () => {
        const { container } = render(
            <NoResultRow showRowWithCardStyle>
                <NoResultCell>Dummy</NoResultCell>
            </NoResultRow>
        );
        expect(container).toMatchSnapshot();
    });
});
