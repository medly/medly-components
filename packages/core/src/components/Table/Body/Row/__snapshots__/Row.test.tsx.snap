// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row should render NoResult properly 1`] = `
.c0 {
  display: grid;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: -webkit-fit-content;
  min-width: -moz-fit-content;
  min-width: fit-content;
  cursor: inherit;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:hover {
  z-index: 2;
}

.c0:nth-child(odd),
.c0:nth-child(odd) > * {
  background-color: #F7F8F9;
}

.c0:nth-child(even),
.c0:nth-child(even) > * {
  background-color: #ffffff;
}

.c0:not(:last-child) {
  border-bottom: 1px solid #C7D0D8;
}

.c1 {
  grid-column: 1/-1;
}

<div>
  <tr
    class="sc-AxjAm c0"
  >
    <td
      class="c1"
    >
      Dummy
    </td>
  </tr>
</div>
`;

exports[`Row should render NoResult with Card design also 1`] = `
.c0 {
  display: grid;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: -webkit-fit-content;
  min-width: -moz-fit-content;
  min-width: fit-content;
  cursor: inherit;
  border: none;
  border-radius: 0.8rem;
  margin-left: 0.8rem;
  margin-right: 0.8rem;
  box-shadow: 0 0.2rem 0.8rem rgba(96,120,144,0.2);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1.5rem 0;
}

.c0,
.c0 > * {
  background-color: #ffffff;
}

.c0.c0.c0 > *:first-child,
.c0.c0.c0 > *:first-child::before {
  -webkit-transition: border-radius 200ms ease-in;
  transition: border-radius 200ms ease-in;
  border-radius: 0.8rem 0 0 0.8rem;
}

.c0.c0.c0 > *:first-child::before {
  width: 0.8rem;
}

.c0.c0.c0 > *:last-child {
  border-radius: 0 0.8rem 0.8rem 0;
}

.c0.c0.c0 > *:nth-last-child(2)::before {
  width: 0.8rem;
  border-radius: 0 0 0 0.8rem;
}

.c0:hover {
  box-shadow: 0 0.2rem 0.8rem rgba(96,120,144,0.35);
}

.c0:not(:last-child) {
  margin-bottom: 0.8rem;
}

.c1 {
  grid-column: 1/-1;
}

<div>
  <tr
    class="sc-AxjAm c0"
  >
    <td
      class="c1"
    >
      Dummy
    </td>
  </tr>
</div>
`;

exports[`Row should render row properly 1`] = `
.c0 {
  display: grid;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: -webkit-fit-content;
  min-width: -moz-fit-content;
  min-width: fit-content;
  cursor: inherit;
}

.c0:hover {
  z-index: 2;
}

.c0:nth-child(odd),
.c0:nth-child(odd) > * {
  background-color: #F7F8F9;
}

.c0:nth-child(even),
.c0:nth-child(even) > * {
  background-color: #ffffff;
}

.c0:not(:last-child) {
  border-bottom: 1px solid #C7D0D8;
}

<div>
  <tr
    class="c0"
  >
    Dummy
  </tr>
</div>
`;
